### ISSUE TITLE
Undo for Deleting a Command Block

### ISSUE THREAD

**Issue by reform-chipper** [wavetermdev/waveterm#421]
As the title says. Currently when a command block is deleted, there's no way to bring it back. There also isn't a confirm dialog box.

The broader goal is to have something like a Jupyter Notebook for shell commands.

Simple example of this vision:

![image](https://github.com/wavetermdev/waveterm/assets/162800380/65c19e44-2917-45ae-a6a4-56d315e697ec)

**Comment by sawka** [comment]
I think a confirm for this might be annoying.  My thought here is to provide some small window of undo (maybe 5 seconds).  So, if you accidentally delete a block you have 5 seconds to "undo it" and it will come back.  This can be implemented by just deferring the actual delete for 5 seconds (although we'll visually remove it immediately).  For the first cut of this feature I'd also only allow one item to be "undone".

I know this isn't a fully general solution, but would this solve enough of the problem for you as a first pass?

**Comment by reform-chipper** [comment]
Yes, that would certainly be better than what we have today. Maybe create an issue to implement a more robust undo system later once there's nothing better to do? 

You're right, a confirm dialog would probably be too intrusive. It might only make sense without the undo feature, because the action is irreversible.
